utils/models.py

```diff
@@ -106,6 +119,23 @@ def get_model(self, model_name: str, extra_args: dict=None, to_compile: bool=Tru
             pass  # Still unsupported :')
 
         return model, diffusion
+    
+    def get_model_name(self, model_name: str, path: str=None) -> str:
+        """
+        Returns the model name extracted from a given path if provided, otherwise returns the original model name.
+
+        Args:
+            model_name (str): The original model name.
+            path (str, optional): The path from which to extract the model name. Defaults to None.
+
+        Returns:
+            str: The extracted model name or the original model name if path is not provided.
+        """
+        if path is None:
+            return model_name
```

Comment:
excuse me, what?

Reply:
If the path is not given the name is detemined to be the name that this class conventionalizes. Maybe I should add a check that model_name is in the dict of the class

Reply:
w/e